File: TestProj/Assets/Titlescreen.js
var delay : float = 0;
var levelIndex : int = 1;
private var timer : float = 0;
function Update(){
	if(timer < delay){
		timer+=Time.deltaTime;
	}
	else{
		if(Input.anyKeyDown){
			Application.LoadLevel(levelIndex);
		}
	}
}

File: TestProj/Assets/Titlescreen.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Titlescreen : MonoBehaviour
{
    public float delay;
    public int levelIndex;
    private float timer;
    public virtual void Update()
    {
        if (this.timer < this.delay)
        {
            this.timer = this.timer + Time.deltaTime;
        }
        else
        {
            if (Input.anyKeyDown)
            {
                Application.LoadLevel(this.levelIndex);
            }
        }
    }

    public Titlescreen()
    {
        this.levelIndex = 1;
    }

}